virtool/samples/api.py

```diff
@@ -680,3 +695,92 @@ async def analyze(req):
             "Location": f"/api/analyses/{analysis_id}"
         }
     )
+
+
+@routes.jobs_api.post("/api/samples/{sample_id}/reads")
+async def upload_reads(req):
+    """
+    Upload sample reads using the Jobs API.
+
+    """
+    db = req.app["db"]
+    pg = req.app["pg"]
+    sample_id = req.match_info["sample_id"]
+
+    reads_file_path = Path(req.app["settings"]["data_path"]) / "samples" / sample_id
+
+    if not await db.samples.find_one(sample_id):
+        return not_found()
+
+    try:
+        if {"reads_1.fq.gz", "reads_2.fq.gz"}.issubset(set(os.listdir(reads_file_path))):
```

Comment:
This check should be done against the database instead of the filesystem.

Reply:
I'm trying to test this properly but for some reason the database is persisting between tests. It's made evident in the [snapshots](https://github.com/virtool/virtool/pull/2139/files#diff-220a2ff3652084488071586d48831973fb1e59235e35ccbab7fb6f0cd8c55968R750-R775) as well, where the `id` should be 1 for each snapshot. It also creates situations like this when running the tests multiple times in succession:
```
>       assert value == snapshot
E       AssertionError: assert {'id': 12, 'n...: 'test', ...} == {'id': 5, 'na...: 'test', ...}
E         Omitting 5 identical items, use -vv to show
E         Differing items:
E         {'id': 12} != {'id': 5}
E         Use -v to get the full diff
```